packages/core/src/shader/UniformGroup.ts

```diff
@@ -8,11 +8,18 @@ let UID = 0;
  */
 export class UniformGroup
 {
+    uniforms: {[key: string]: any};
+    group: boolean;
```

Comment:
These should be readonly.

Also is there a better type than any for this?

Reply:
Its either UniformGroup either number either Float32Array or Uint16Array or a texture... we can think about it